src/dune_rules/lib.mli

```diff
@@ -160,6 +160,21 @@ module DB : sig
     -> dune_version:Dune_lang.Syntax.Version.t
     -> Compile.t
 
+  (** Resolve libraries written by the user in a [dune] file. The resulting list
+      of libraries is transitively closed and sorted by the order of
+      dependencies.
+
+      This function is for melange.emit stanzas. *)
+  val resolve_user_written_deps_for_melange :
```

Comment:
This function has the exact same signature as the other one. So instead of introducing it, why don't we just modify the old function to accept one more parameter to distinguish exe and emit.

Reply:
Fixed in https://github.com/ocaml/dune/pull/6454/commits/0b7f398c2184e5109de9055007cf502c9f356198, this adds some extra code on callers to prepare `melange_ident` but I assume this is preferable.